src/shadowbox/server/manager_service.ts

```diff
@@ -210,6 +223,14 @@ export class ShadowsocksManagerService {
   public renameAccessKey(req: RequestType, res: ResponseType, next: restify.Next): void {
     try {
       logging.debug(`renameAccessKey request ${JSON.stringify(req.params)}`);
+      if (!isRequestParam(req.params, 'id')) {
+        return next(
+            new restify.MissingParameterError({statusCode: 400}, 'Parameter `id` is missing'));
+      }
+      if (!isRequestParam(req.params, 'name')) {
+        return next(
+            new restify.MissingParameterError({statusCode: 400}, 'Parameter `name` is missing'));
+      }
       const accessKeyId = req.params.id;
       this.accessKeys.renameAccessKey(accessKeyId, req.params.name);
```

Comment:
As an example of the kind of thing I'm worried about: even with the new checks, there's nothing here that stops the caller from passing a non-`string` value for `req.params.name`, which could have all kinds of weird effects.